lib/ex_json_path.ex

```diff
@@ -24,7 +24,15 @@ defmodule ExJsonPath do
   end
 
   def eval(map, compiled_path) when is_map(map) and is_list(compiled_path) do
-    recurse(map, compiled_path)
+    result = recurse(map, compiled_path)
```

Comment:
There's one case where you return `{:error, :no_match}` in `recurse`, so this should probably go in a `with` block